pymc3/sampling.py

```diff
@@ -433,20 +426,18 @@ def sample(
     if cores is None:
         cores = min(4, _cpu_count())
 
+    if not isinstance(random_seed, (Iterable, int)):
+        raise TypeError("Invalid value for `random_seed`. Must be tuple, list or int")
     if chains is None:
         chains = max(2, cores)
     if isinstance(start, dict):
         start = [start] * chains
-    if random_seed == -1:
-        random_seed = None
```

Comment:
Moving this into line 435, as it looks like `random_seed=-1` is meant to be treated the same way as `random_seed=None`